src/libasr/pass/for_all.cpp

```diff
@@ -36,7 +36,7 @@ class ForAllVisitor : public ASR::StatementWalkVisitor<ForAllVisitor>
         heads.reserve(al,1);
         heads.push_back(al, x.m_head);
         ASR::stmt_t *stmt = ASRUtils::STMT(
-            ASR::make_DoConcurrentLoop_t(al, loc, heads.p, heads.n, nullptr, 0, nullptr, 0, nullptr, 0, body.p, body.size())
+            ASR::make_DoConcurrentLoop_t(al, loc, heads.p, heads.n, nullptr, 0, nullptr, 0, nullptr, 0, body.p, body.size(), 1)
```

Comment:
```suggestion
            ASR::make_DoConcurrentLoop_t(al, loc, heads.p, heads.n, nullptr, 0, nullptr, 0, nullptr, 0, body.p, body.size(), 0)
```

If no collapse, then make it `0`.